cpp/build-support/convert-benchmark.py

```diff
@@ -0,0 +1,299 @@
+        aggs, runs = partition(lambda b: b.is_agg, runs)
+        self.runs = sorted(runs, key=lambda b: b.value)
+        self.values = [b.value for b in self.runs]
+        self.aggregates = aggs
+
+    @property
+    def n_runs(self):
+        return len(self.runs)
+
+    @property
+    @memoize
+    def mean(self):
+        maybe_mean = [b for b in self.aggregates if b.is_mean]
+        if maybe_mean:
+            return maybe_mean[0].value
+        # fallback
+        return sum(self.values) / self.n_runs
+
+    @property
+    @memoize
+    def median(self):
+        maybe_median = [b for b in self.aggregates if b.is_median]
+        if maybe_median:
+            return maybe_median[0].value
+        # fallback
+        return self.runs[int(self.n_runs / 2)].value
+
+    @property
+    @memoize
+    def stddev(self):
+        maybe_stddev = [b for b in self.aggregates if b.is_stddev]
+        if maybe_stddev:
+            return maybe_stddev[0].value
+
+        sum_diff = sum([(val - self.mean)**2 for val in self.values])
+        return (sum_diff / (self.n_runs - 1))**0.5 if self.n_runs > 1 else 0.0
+
+    @property
+    def min(self):
+        return self.values[0]
+
+    @property
+    def max(self):
+        return self.values[-1]
+
+    def quartile(self, q):
+        return self.values[int(q * self.n_runs / 4)]
+
+    @property
+    def q1(self):
+        return self.quartile(1)
+
+    @property
+    def q3(self):
+        return self.quartile(3)
+
+    @property
+    def parameters(self):
+        """ Extract parameters from Benchmark's name"""
+        def parse_param(idx, param):
+            k_v = param.split(":")
+            # nameless parameter are transformed into positional names
+            name = k_v[0] if len(k_v) > 1 else f"arg{idx}"
+            return name, k_v[-1]
+
+        params = enumerate(self.name.split("/")[1:])
+        named_params = [parse_param(idx, p) for idx, p in params if p]
+        return {k: v for k, v in named_params}
+
+    def as_arrow(self):
+        n_runs = self.n_runs
+        run = {
+            "benchmark_name": self.name,
+            "value": self.mean,
+            "val_min": self.min,
+            "val_q1": self.q1,
+            "median": self.median,
+            "val_q3": self.q3,
+            "val_max": self.max,
+            "std_dev": self.stddev,
+            "n_obs": n_runs,
+        }
+
+        params = self.parameters
+        if params:
+            run["parameter_values"] = params
+
+        return run
+
+    def __str__(self):
+        return f"BenchmarkSuite[name={self.name},runs={len(self.runs)}]"
+
+    @classmethod
+    def from_json(cls, version, payload):
+        def group_key(x):
+            return x.name
+
+        benchmarks = map(lambda x: BenchmarkObservation(version, **x), payload)
+        groups = groupby(sorted(benchmarks, key=group_key), group_key)
+        return [cls(k, version, list(bs)) for k, bs in groups]
+
+
+def as_arrow(version, payload):
```

Comment:
Would be nice to add a comment or docstring somewhere explaining what this script does. It will make maintenance easier in 6 months ;-)